components/indexer.py

```diff
@@ -130,46 +133,35 @@ def execute(self) -> None:
 
         # Override any limit switches where the next cell is vacant
         for first, second in zip(self.all_motors, self.all_motors[1:]):
-            at_limit = second.isFwdLimitSwitchClosed()
-            if second == feeder and self.transfer_to_injector:
-                at_limit = True  # Pretend the ball is still in the feeder
-            if not at_limit:
-                first.overrideLimitSwitchesEnable(False)
-            else:
-                first.overrideLimitSwitchesEnable(True)
-
-        if self.intaking:
-            if not intake_main_motor.isFwdLimitSwitchClosed():
-                if self.shimmying:
-                    if self.left_shimmy:
-                        self.intake_left_motor.set(self.shimmy_speed)
-                        self.intake_right_motor.set(0)
-                        self.shimmy_count += 1
-                        if self.shimmy_count > self.shimmy_ticks:
-                            self.left_shimmy = False
-                            self.shimmy_count = 0
-                    else:
-                        self.intake_left_motor.set(0)
-                        self.intake_right_motor.set(self.shimmy_speed)
-                        self.shimmy_count += 1
-                        if self.shimmy_count > self.shimmy_ticks:
-                            self.left_shimmy = True
-                            self.shimmy_count = 0
-                else:
+            second_has_ball = second.isFwdLimitSwitchClosed()
+            if second is feeder and self.transfer_to_injector:
+                second_has_ball = True  # Pretend the ball is still in the feeder
+            first.overrideLimitSwitchesEnable(second_has_ball)
+
+        if self.intaking and not intake_main_motor.isFwdLimitSwitchClosed():
+            if self.shimmying:
+                if self.left_shimmy:
                     self.intake_left_motor.set(self.shimmy_speed)
+                    self.intake_right_motor.set(0)
+                    self.shimmy_count += 1
+                    if self.shimmy_count > self.shimmy_ticks:
+                        self.left_shimmy = False
+                        self.shimmy_count = 0
+                else:
+                    self.intake_left_motor.set(0)
                     self.intake_right_motor.set(self.shimmy_speed)
-
+                    self.shimmy_count += 1
+                    if self.shimmy_count > self.shimmy_ticks:
+                        self.left_shimmy = True
+                        self.shimmy_count = 0
             else:
-                self.intake_right_motor.set(0)
-                self.intake_left_motor.set(0)
-        else:
+                self.intake_left_motor.set(self.shimmy_speed)
+                self.intake_right_motor.set(self.shimmy_speed)
```

Comment:
This is a functional rather than stylistic change, not sure if we ever want to run them simultaneously due to jam risks

Reply:
This has exactly the same behaviour as previously - notice which if statement this is for.

Reply:
I see, still not sure we want to do this but it is the only way the intake functions without shimmying.